zerver/lib/presence.py

```diff
@@ -151,23 +151,36 @@ def get_presence_dict_by_realm(
     realm: Realm,
     slim_presence: bool = False,
     last_update_id_fetched_by_client: int | None = None,
+    history_limit_days: int | None = None,
     requesting_user_profile: UserProfile | None = None,
 ) -> tuple[dict[str, dict[str, Any]], int]:
-    two_weeks_ago = timezone_now() - timedelta(weeks=2)
+    now = timezone_now()
+    if history_limit_days is not None:
+        fetch_since_datetime = now - timedelta(days=history_limit_days)
+    else:
+        # The original behavior for this API was to return last two weeks
+        # of data at most, so we preserve that when the history_limit_days
+        # param is not provided.
+        fetch_since_datetime = now - timedelta(days=14)
```

Comment:
Hmm, but how does a client request "all users" with this construction?

Reply:
Hmm, my reading around https://chat.zulip.org/#narrow/stream/378-api-design/topic/presence.20data.20cutoffs/near/1921788 was that for now clients can just pass a large value for `history_limit_days` for "fetch everything" for now - and maybe we make that the default when `history_limit_days=None` later. 

Added a tweak to eliminate the `last_connected_time` clause in the query when a non-initial value for `last_update_id` is passed: